mipengine/controller/algorithm_execution_DTOs.py

```diff
@@ -12,6 +13,7 @@ class AlgorithmExecutionDTO(BaseModel):
     context_id: str
     algorithm_name: str
     algorithm_request_dto: AlgorithmRequestDTO
+    datasets_per_local_node: Dict[str, List[str]]
```

Comment:
having the datasets both in the datasets_per_local_node as well as in the algorithm_request_dto constitutes duplication and it will very likely cause problems in the future.

Reply:
I fixed this by removing the `algorithm_request_dto` from the `AlgorithmExecutionDTO` and adding directly the vars needed inside.

From looking a bit further down the code, I saw the `AlgorithmExecutionInterfaceDTO` that is very similar to the `AlgorithmExecutionDTO`. Maybe they should be merged into one. Any additional parameters could be provided as separate arguments. Just some thoughs, not for this PR.